frontend/packages/metal3-plugin/src/plugin.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  Plugin,
+  ResourceNSNavItem,
+  ResourceListPage,
+  ResourceDetailPage,
+  ModelFeatureFlag,
+  ModelDefinition,
+} from '@console/plugin-sdk';
+
+import { BaremetalHostModel } from './models';
+
+type ConsumedExtensions =
+  | ResourceNSNavItem
+  | ResourceListPage
+  | ResourceDetailPage
+  | ModelFeatureFlag
+  | ModelDefinition;
+
+const METAL3_FLAG = 'METAL3'
+
+const plugin: Plugin<ConsumedExtensions> = [
+  {
+    type: 'ModelDefinition',
+    properties: {
+      models: [BaremetalHostModel],
+    },
+  },
+  {
+    type: 'FeatureFlag/Model',
+    properties: {
+      model: BaremetalHostModel,
+      flag: METAL3_FLAG
+    },
+  },
+  {
+    type: 'NavItem/ResourceNS',
+    properties: {
+      section: 'Compute',
+      componentProps: {
+        name: 'Bare Metal Hosts',
+        resource: 'baremetalhosts',
+        required: METAL3_FLAG
+      },
```

Comment:
If you want to put your nav link after an existing one, you can use `mergeAfter`.

Reply:
Done